access/slackbot/slackbot.go

```diff
@@ -273,8 +272,19 @@ func (a *App) Start(ctx context.Context) error {
 	go func() {
 		defer shutdown()
 		defer workers.Done()
-		if err := a.watchRequests(ctx); err != nil {
-			watcherErr = trace.Wrap(err)
+		for {
+			log.Info("Starting a request watcher...")
+			if err := a.watchRequests(ctx); err != nil {
+				if trace.IsEOF(err) || trace.IsConnectionProblem(err) {
+					log.Error("Watcher connection closed: ", err)
+					continue
```

Comment:
I think a backoff mechanism is warranted here, could be something simple like a linearly increasing sleep which resets after a short cooldown.

Reply:
grpc library has its internal backoff implementation. I configure it via `grpc.WithBackoffMaxDelay(time.Second*2)` option.

With `WaitForReady(false)` it has no effect but combining it with `WaitForReady(true)` this loop would block on next iteration until the server is up.